pkg/rules/engine/engine.go

```diff
@@ -324,41 +307,23 @@ func (engine *Engine) UnloadSignature(signatureId string) error {
 	return nil
 }
 
-// ParsedEvent holds the original tracee.Event and its OPA ast.Value representation.
-type ParsedEvent struct {
-	Event external.Event
-	Value ast.Value
-}
-
-// ToParsedEvent enhances tracee.Event with OPA ast.Value. This is mainly used
-// for performance optimization to avoid parsing tracee.Event multiple times.
-func ToParsedEvent(e external.Event) (ParsedEvent, error) {
-	u, err := e.ToUnstructured()
-	if err != nil {
-		return ParsedEvent{}, fmt.Errorf("unstructuring event: %w", err)
-	}
-	// TODO In OPA >= v0.30.0 we can try passing tracee.Event directly to get rid of ToUnstructured call.
-	v, err := ast.InterfaceToValue(u)
-	if err != nil {
-		return ParsedEvent{}, fmt.Errorf("converting unstructured event to OPA ast.Value: %w", err)
+func eventSignatureSelector(event protocol.Event) (detect.SignatureEventSelector, error) {
+	origin := event.Origin()
```

Comment:
Is is still in scope of event header refactor? Looks to me like we are changing the logic and way event selectors are used.

Reply:
this conversation can be resolved IMO @danielpacak

Reply:
Okey. Thank you for the confirmation.